### PR TITLE
Ensure Children component order is maintained

### PR DESCRIPTION
**Pull request by matthewroach**
## Description

Update the Price Component to ensure the children components that are only allowed to be used are output in the order passed into the component

## Test Steps

* Tests have been updated

1. Checkout branch - `git checkout price-component-maintain-child-order`
2. Update dependencies - `npm i`
3. Run Storybook `npm run storybook`
4. Check out the Price storybook documentation

## Author Checklist

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed relevant documentation has been updated/added.
- [x] Add label - **ready for review** when the pull request is ready for someone to begin reviewing

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] All GitHub Actions pass
- [ ] Ran the code locally based on the test instructions.
- [ ] Checked Chromatic for Storybook changes, accepted the updates if acceptable
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.
- [ ] Approve and Add label - **ready to merge** if you are happy with the pull request
- [ ] Want another reviewer? Add the label **additional review**

### CODE REVIEW

File: src/components/price/index.jsx [hunk 1]
```diff
@@ -6,8 +6,9 @@ import Sale from "./_children/Sale";
 const COMPONENT_CLASS_NAME = "c-price";
 
 const Price = ({ children, className, ...rest }) => {
-	const subComponents = Object.values(Price).map((subcomponentType) =>
-		Children.map(children, (child) => (child?.type === subcomponentType ? child : null))
+	const allowedTypes = Object.values(Price).map((subcomponentType) => subcomponentType);
```
Comment (brentmiller1973) [inline src/components/price/index.jsx:10]: I think these two lines would benefit future developers with just a line or two of comments explaining how this is ensuring children render order.

**Review by brentmiller1973 — CHANGES_REQUESTED** [review]
@matthewroach Brilliant, I just would like a code comment in index.jsx to provide immediate clarity to why this is being done.

**Review by JackHowa — APPROVED** [review]
Makes sense

File: src/components/price/index.jsx (hunk 1 above)
Comment (matthewroach) [reply src/components/price/index.jsx:10]: Add comments

Comment (brentmiller1973) [reply src/components/price/index.jsx:10]: Very nice

**Review by brentmiller1973 — APPROVED** [review]

**Pull request merged**